ambry-store/src/main/java/com/github/ambry/store/CompactAllPolicyFactory.java

```diff
@@ -58,13 +61,28 @@ class CompactAllPolicy implements CompactionPolicy {
 
   @Override
   public CompactionDetails getCompactionDetails(long totalCapacity, long usedCapacity, long segmentCapacity,
-      long segmentHeaderSize, List<LogSegmentName> logSegmentsNotInJournal, BlobStoreStats blobStoreStats, String dataDir) {
+      long segmentHeaderSize, List<LogSegmentName> logSegmentsNotInJournal, BlobStoreStats blobStoreStats,
+      String dataDir) throws StoreException {
     CompactionDetails details = null;
     logger.trace("UsedCapacity {} vs TotalCapacity {}", usedCapacity, totalCapacity);
     if (usedCapacity >= (storeConfig.storeMinUsedCapacityToTriggerCompactionInPercentage / 100.0) * totalCapacity) {
       if (logSegmentsNotInJournal != null) {
         details = new CompactionDetails(time.milliseconds() - messageRetentionTimeInMs, logSegmentsNotInJournal, null);
         logger.info("Generating CompactionDetails {} using CompactAllPolicy", details);
+
+        Pair<Long, NavigableMap<LogSegmentName, Long>> validDataSizeByLogSegment =
+            blobStoreStats.getValidDataSizeByLogSegment(
+                new TimeRange(time.milliseconds() - messageRetentionTimeInMs - ERROR_MARGIN_MS, ERROR_MARGIN_MS));
+        if (validDataSizeByLogSegment != null) {
+          final StringBuilder sizeLog = new StringBuilder(
+              "Valid data size for " + dataDir + " from BlobStoreStats " + validDataSizeByLogSegment.getFirst()
+                  + " segments: ");
+          validDataSizeByLogSegment.getSecond().forEach((logSegmentName, validDataSize) -> {
+            sizeLog.append(logSegmentName + " " + validDataSize / 1000 / 1000 / 1000.0 + "GB "
+                + validDataSize * 1.0 / segmentCapacity + "%;");
```

Comment:
Yes, it's good to have file size or in-memory total size(not valid size) data. 
I try to avoid system call to get the file size for debug purpose. I don't find a in-memory one handy. Let me know if I miss anything.